Created KingdomBinaryPackage, MobilePartyBinaryPackage, and ArmyBinaryPackage. Also two binary packages that are non-issue, ExplainedNumberBinaryPackage and PathFaceRecordBinaryPackage.

zzzzzzzzott opened the pull request:
## PR Checklist
<!-- Insert "x" inside square brackets to check item. -->

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:


## What is the current behaviour?
<!-- Please describe the current behaviour that you are modifying, or link to a relevant issue. -->


## What is the new behaviour?
<!-- Insert issue id after hashtag. -->
Resolves #393 #366 #381 
<!-- Describe functionality added. Add images or videos to show how it works if applies -->


## Other information
Also added ExplainedNumberBinaryPackage and PathFaceRecordBinaryPackage as they were needed for MobilePartyBinaryPackage.

zzzzzzzzott committed:
Resolves #393, #366, #381

Also introduces non-issue binary packages - ExplainedNumber and PathFaceRecord.

garrettluskey commented on source/GameInterface/Serialization/Impl/ExplainedNumberBinaryPackage.cs:
```diff
@@ -0,0 +1,41 @@
+namespace GameInterface.Serialization.Impl
+{
+    /// <summary>
+    /// Binary package for Kingdom
```
Name is wrong here

garrettluskey commented on source/GameInterface/Serialization/Impl/PathFaceRecordBinaryPackage.cs:
```diff
@@ -0,0 +1,41 @@
+namespace GameInterface.Serialization.Impl
+{
+    /// <summary>
+    /// Binary package for Kingdom
```
```suggestion
    /// Binary package for PathFaceRecord
```

garrettluskey committed:
Cleaned up MobileParty and Kingdom

Added new Bootstrap capability and reorganized it's folder structure

Cleaned up MobileParty serialization and test

Cleaned up Kingdom

garrettluskey committed:
Cleanup

garrettluskey committed:
Merge branch 'taverns-serialization' into pr/425

garrettluskey committed:
Army cleanup
